### PR TITLE
MEN-1421 limits and internal PUT endpoint

### PR DESCRIPTION
**Pull request by bboozzoo**
@mendersoftware/rndity @maciejmrowiec

### CODE REVIEW

File: store/datastore.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ var (
 	ErrTokenNotFound = errors.New("token not found")
 	// authorization set not found
 	ErrAuthSetNotFound = errors.New("authorization set not found")
+	// authorization set not found
```
Comment (mchalski) [inline store/datastore.go]: limit

Comment (bboozzoo) [reply store/datastore.go]: fixed

**Comment by GregorioDiStefano** [comment]
I have a fix for the spurious test fail, I restarted this test in the meantime

**Comment by bboozzoo** [comment]
thanks

**Comment by bboozzoo** [comment]
failed in MT artifact upload again:
```
self = <MenderAPI.deployments.Deployments object at 0x7fdd53b570d0>
filename = '/tmp/tmpvzyzcI', description = 'abc'

def upload_image(self, filename, description="abc"):
image_path_url = self.get_deployments_base_path() + "artifacts"

r = requests.post(image_path_url,
verify=False,
headers=self.auth.get_auth_token(),
files=(
("description", (None, description)),
("size", (None, str(os.path.getsize(filename)))),
("artifact", (filename, open(filename), "application/octet-stream"))
))

logger.info("Received image upload status code: " + str(r.status_code) + " with payload: " + str(r.text))
> assert r.status_code == requests.status_codes.codes.created
E       AssertionError
```
restarted

**Comment by bboozzoo** [comment]
merging

**Pull request merged**